src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,127 +1,109 @@
 
+import java.time.DateTimeException;
+import java.time.Instant;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Locale;
 import java.util.Optional;
 
 public class JavaDateTimeApi {
-    /**
-     * Верните текущую дату в виде строки в зависимости от запроса.
-     *
-     * @param datePart Запрос на часть даты или всю дата целиком:
-     *                 - FULL - текущая дата целиком год, месяц, день (число месяца)
-     *                 в формате YYYY-MM-DD, возвращаемое значение по умолчанию;
-     *                 - YEAR - текущий год;
-     *                 - MONTH - название текущего месяца;
-     *                 - DAY - текущий день (число месяца);
-     *                 В любом другом случае бросить DateTimeException
-     **/
+    private static final int FIRST_INDEX = 0;
+    private static final int SECOND_INDEX = 1;
+    private static final int THIRD_INDEX = 2;
+    private static final ZoneOffset UKRAINE_TIMEZONE = ZoneOffset.of("+02:00");
+    private static final DateTimeFormatter DATE_FORMATTER
+            = DateTimeFormatter.ofPattern("d MMM yyyy", Locale.ENGLISH);
+    private static final DateTimeFormatter DATE_TIME_FORMATTER
+            = DateTimeFormatter.ofPattern("dd LLLL yyyy HH:mm", Locale.ENGLISH);
+
     public String todayDate(DateTimePart datePart) {
-        return "Today";
+        LocalDate now = LocalDate.now();
+        String currentDate;
+        switch (datePart) {
+            case FULL:
+                currentDate = now.toString();
+                break;
+            case DAY:
+                currentDate = String.valueOf(now.getDayOfMonth());
+                break;
+            case YEAR:
+                currentDate = String.valueOf(now.getYear());
+                break;
+            case MONTH:
+                currentDate = String.valueOf(now.getMonthValue());
+                break;
+            default:
+                throw new DateTimeException("It's not correct data/time format");
+        }
+        return currentDate;
     }
 
-    /**
-     * Верните Optional даты соответствующей дате в массиве.
-     *
-     * @param dateParams Дан массив данных состоящий из 3-х элементов, где:
-     *                   - 1-й элемент массива - год;
-     *                   - 2-й элемент массива - месяц;
-     *                   - 3-й элемент массива - день (число);
-     */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        try {
+            return Optional.of(LocalDate.of(dateParams[FIRST_INDEX],
+                    dateParams[SECOND_INDEX], dateParams[THIRD_INDEX]));
+        } catch (DateTimeException | ArrayIndexOutOfBoundsException e) {
+            return Optional.empty();
```

Comment:
we can catch two particular exceptions in the same catch block, but in case of any trouble we should be able to identify them - for now you can use System.out.println(), later we will learn logging

move return Optional.empty(); to the end of method